### PR TITLE
[dvsim] Small dvsim reporting updates

### PR DESCRIPTION
**Pull request by msfschaffner**
This adds two small reporting updates to dvsim:

1. additional fields are added to the flow class in order to selectively sanitize email and published reports

2. the git commit and branch information is now displayed in the reports, right after the timestamp

### CODE REVIEW

File: util/dvsim/FlowCfg.py [hunk 1]
```diff
@@ -83,10 +88,16 @@ def __init__(self, flow_cfg_file, proj_root, args):
         self.results_summary_server_html = ""
         self.results_summary_server_page = ""
 
-        # Full and summary results in md text.
+        # Full results in md test
```
Comment (cindychip) [inline util/dvsim/FlowCfg.py:87]: test? I thought text makes sense to me..

File: hw/syn/data/dc.hjson [hunk 2]
```diff
@@ -26,5 +26,8 @@
                 "--outdir {build_dir}"]
 
   // Restrict the maximum message count to 10 in each category
-  max_msg_count: 10
+  max_msg_count: 20
```
Comment (cindychip) [inline hw/syn/data/dc.hjson:29]: you might want to change the comment in line 28 as well to `20`

**Comment by msfschaffner** [comment]
Ok, I've modified the commit/branch info feature such that the revision string is entirely assembled on the user side (inside an `{eval_cmd}` in the project hjson config).

File: util/dvsim/FlowCfg.py (hunk 1 above)
Comment (msfschaffner) [reply util/dvsim/FlowCfg.py:87]: wuups... typo

File: hw/syn/data/dc.hjson (hunk 2 above)
Comment (msfschaffner) [reply hw/syn/data/dc.hjson:29]: right, thanks!

**Comment by msfschaffner** [comment]
> Hi Michael, thanks for your commits. The sanitize email/publish results one are very useful for FPV as well. Appreciate your contribution! I just want to make sure I understand:
> The sanitize email only works for individual CFGs, not for a master CFG? For master cfg, we are still using the `email_summary_md`? (If that is the case, maybe I can work on unify the naming)
> But LGTM.

Yes that is right. I've added it to the detailed reports since that is where lint and synthesis expose the sensitive info. 
However, if you think this could be useful to have for summary reports for FPV for instance, it would be great if you could extend this.

File: hw/syn/data/dc.hjson [hunk 3]
```diff
@@ -25,6 +25,9 @@
                 "--reppath {build_dir}/REPORTS/ ",
                 "--outdir {build_dir}"]
 
-  // Restrict the maximum message count to 10 in each category
-  max_msg_count: 10
+  // Restrict the maximum message count in each category
+  max_msg_count: 20
+  // Do not sanitize regression mails, but do sanitize the published report
+  sanitize_email_results:   "False"
```
Comment (sriyerg) [inline hw/syn/data/dc.hjson:31]: Looks like HJson parses `true`, `false` to boolean: https://hjson.github.io/syntax.html

File: util/dvsim/FlowCfg.py [hunk 4]
```diff
@@ -83,10 +84,16 @@ def __init__(self, flow_cfg_file, proj_root, args):
         self.results_summary_server_html = ""
         self.results_summary_server_page = ""
 
-        # Full and summary results in md text.
+        # Full results in md text
         self.results_md = ""
+        # Selectively sanitized md results to be mailed out or published
+        self.email_results_md = ""
+        self.publish_results_md = ""
+        self.sanitize_email_results = ""
```
Comment (sriyerg) [inline util/dvsim/FlowCfg.py:92]: I think these are better initialized to `False` (once you make the HJson change above).

File: hw/syn/data/dc.hjson (hunk 3 above)
Comment (msfschaffner) [reply hw/syn/data/dc.hjson]: cool, let me change that to bools.

File: util/dvsim/FlowCfg.py (hunk 4 above)
Comment (msfschaffner) [reply util/dvsim/FlowCfg.py]: jep, done.

**Pull request merged**